src/o365/spo/commands/search/search.ts

```diff
@@ -0,0 +1,434 @@
+
+    //Construct single requestUrl
+    const requestUrl = `${webUrl}/_api/search/query?querytext='${args.options.query}'`.concat(
+      propertySelectRequestString,
+      startRowRequestString,
+      rowLimitRequestString,
+      sourceIdRequestString,
+      trimDuplicatesRequestString,
+      enableStemmingRequestString,
+      cultureRequestString,
+      refinementFiltersRequestString,
+      queryTemplateRequestString,
+      sortListRequestString,
+      rankingModelIdRequestString,
+      propertiesRequestString,
+      refinersRequestString,
+      hiddenConstraintsRequestString,
+      clientTypeRequestString,
+      enablePhoneticRequestString,
+      processBestBetsRequestString,
+      enableQueryRulesRequestString,
+      processPersonalFavoritesRequestString
+    );
+
+    if(this.debug) {
+      cmd.log(`RequestURL: ${requestUrl}`);
+    }
+    return requestUrl;
+  }
+
+  private getPropertiesRequestString(args: CommandArgs):string {
+    let properties = args.options.properties ? args.options.properties : '';
+    if(args.options.sourceName) {
+      if(properties && !properties.endsWith(",")) { properties += `,`; }
+      properties += `SourceName:${args.options.sourceName},SourceLevel:SPSite`;
+    }
+    return properties ? `&properties='${properties}'` : ``;
+  }
+
+  private getSelectPropertiesArray(args: CommandArgs) {
+    return args.options.selectProperties 
+      ? args.options.selectProperties.split(",")
+      : (!args.options.output || args.options.output === "text") 
+        ? ["Rank","DocId","OriginalPath"/*,"PartitionId","UrlZone","Culture","ResultTypeId","RenderTemplateId"*/]
+        : [];
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-q, --query <query>',
+        description: 'Query to be executed in KQL format'
+      },
+      {
+        option: '-p, --selectProperties <selectProperties>',
+        description: 'Comma separated list of properties to retrieve. Will retrieve all properties if not specified and json output is requested.'
+      },
+      {
+        option: '--allResults',
+        description: 'Get all results of the search query, not only the amount specified by the rowlimit (default: 10)'
+      },
+      {
+        option: '--rowLimit <rowLimit>',
+        description: 'Sets the number of rows to be returned. When the \'allResults\' parameter is enabled, it will determines the size of the batches being retrieved'
+      },
+      {
+        option: '--sourceId <sourceId>',
+        description: 'Specifies the identifier (GUID) of the result source to be used to run the query.'
+      },
+      {
+        option: '--trimDuplicates',
+        description: 'Specifies whether near duplicate items should be removed from the search results.'
```

Comment:
Let's rephrase to `Set, to remove near duplicate items from the search results`. Because you can't specify a value and just set the option or not, it would be more clearer to have a description that reflects it. The same applies to other switch-options in the command.